### DISCUSSION TITLE
Token Authentication Requirement for Git Operations

### DISCUSSION THREAD

**Discussion by jeet-khondker** [community/community#23919] [New to GitHub]
<p>Today is August 14, 2021.</p>
<p>I was working on a very important long-term project and I used the command <code>git add .</code> to add my commits. But when I enter the command <code>git push</code> it gives me the following error:</p>
<pre><code class="lang-auto">remote: Support for password authentication was removed on August 13, 2021. Please use a personal access token instead.
remote: Please see https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/ for more information.
fatal: unable to access 'https://github.com/jeet-khondker/repository-name.git/': The requested URL returned error: 403
</code></pre>
<p>Basically I am using VSCode to execute my Git operations.</p>
<p>So when I look over to the blog mentioned in the error message, I found that I need to use token authentication for Git operations instead of password authentication.</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/48243450/181080014-2b7b098e-0e95-4656-bc47-a7a6568a2164.png" class="site-icon" width="32" height="32">

      <a href="https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/" target="_blank" rel="noopener nofollow ugc" title="08:28PM - 15 December 2020">The GitHub Blog – 15 Dec 20</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://user-images.githubusercontent.com/48243450/181080015-cbcf0d9a-901f-40c2-8744-60ad1d7e2d0b.png" class="thumbnail" width="690" height="362" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/2/c25410e336a7f8815e23b481139793f4faef75d1_2_690x362.png, https://aws1.discourse-cdn.com/github/optimized/3X/c/2/c25410e336a7f8815e23b481139793f4faef75d1_2_1035x543.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/c/2/c25410e336a7f8815e23b481139793f4faef75d1.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/2/c25410e336a7f8815e23b481139793f4faef75d1_2_10x10.png"></div>

<h3><a href="https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/" target="_blank" rel="noopener nofollow ugc">Token authentication requirements for Git operations | The GitHub Blog</a></h3>

  <p>Beginning August 13th, 2021, we will no longer accept account passwords when authenticating Git operations on GitHub.com.</p>

  <p>
    <span class="label1">Est. reading time: 3 minutes</span>
  </p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>I am kind of afraid how to overcome this issue as because this is a very important project I am working on and I don’t want to loose any of my previous code commits.</p>
<p>May I have your kind support of how to overcome this issue and switch from password authentication to token authentication without disrupting my codebase?</p>
<p>Thank you.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>You need to create <a href="https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token">Create a PAT</a> as described in the blog post, and use that instead of your password for pushing (or other authenticated Git operations). If the password is cached somewhere, you might need to delete that.</p>
<p>The PAT should probably have the <strong>repo</strong> and <strong>workflow</strong> scopes. The first gives access to all your repositories, the second lets you modify GitHub Actions workflows.</p>

**Comment by jeet-khondker** [comment]
<p>Hi Luna,</p>
<p>Thank you for your response to my problem. I finally solved my issue.</p>
<p>Thanking you once again.</p>

**Comment by glenulmer** [comment]
<p>As best I can understand the PAT installation process described, I’ve followed the same process on two different linux systems. On one, everything works as expected. On the other, I am blocked from using https operations.</p>
<p>Output from <code>git config --list</code> is exactly the same on both systems for user.name, user.email and user.password.  There are some other differences between the two git config listings, but as far as I understand it, user.name and user.password are the only relevant ones.</p>
<p>As airtower writes, it may be a caching issue. But I have no idea how to clear the cache.</p>
<p>Are there other git config settings I’ve missed? Why else might one system work and not the other?</p>
<p>I appreciate any specific steps to take to diagnose or resolve the problem. Thanks.</p>

**Comment by jeet-khondker** [comment]
<p>I am using Mac OS, and I successfully solved my issue using Keychain Access.</p>
